When braille mode is set to follow cursors, the image description is not shown in braille and an error is logged

### Brief summary

_No response_

### Steps to reproduce

* Use an actual braille display or the braille viewer, and make sure the AI image description feature is downloaded and works. Also make sure you have set the braille mode to follow cursors.
* press NVDA+Windows+comma to describe the navigator object.

### Actual behavior

The description is spoken, but not shown in braille. The following error is logged.
ERROR - extensionPoints.Action.notify (23:09:05.185) - RunCaptionThread (10412):
Error running handler <bound method BrailleViewerFrame.updateBrailleDisplayed of <brailleViewer.brailleViewerGui.BrailleViewerFrame object at 0x0000020D9249D9A0>> for <extensionPoints.Action object at 0x0000020D81764D70>
Traceback (most recent call last):
  File "extensionPoints\__init__.pyc", line 56, in notify
  File "extensionPoints\util.pyc", line 230, in callWithSupportedKwargs
  File "brailleViewer\brailleViewerGui.pyc", line 473, in updateBrailleDisplayed
  File "brailleViewer\brailleViewerGui.pyc", line 482, in _triggerGuiUpdate
wx._core.wxAssertionError: C++ assertion "wxThread::IsMain()" failed at ..\..\src\common\timerimpl.cpp(57) in wxTimerImpl::Start(): timer can only be started from the main thread
ERROR - logHandler._threadExceptHook (23:09:05.197) - RunCaptionThread (10412):
Exception in thread RunCaptionThread:
Traceback (most recent call last):
  File "threading.pyc", line 1043, in _bootstrap_inner
  File "threading.pyc", line 994, in run
  File "_localCaptioner\imageDescriber.pyc", line 81, in _messageCaption
  File "ui.pyc", line 251, in message
  File "braille.pyc", line 2979, in message
  File "braille.pyc", line 2991, in _resetMessageTimer
  File "wx\core.pyc", line 3492, in Start
wx._core.wxAssertionError: C++ assertion "wxThread::IsMain()" failed at ..\..\src\common\timerimpl.cpp(57) in wxTimerImpl::Start(): timer can only be started from the main thread

### Expected behavior

The description is shown in braille as well

### NVDA logs, crash dumps and other attachments

_No response_

### NVDA type

installer/launcher copy

### NVDA version

nvda_snapshot_alpha-52937,f535af1f.exe

### Have you tried any other versions of NVDA? If so, please report their behaviors.

No

### Windows version

n/a

### Name and version of other software in use when reproducing the issue

n/a

### Other information about your system

n/a

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

--- a/source/_localCaptioner/captioner.py
+++ b/source/_localCaptioner/captioner.py
@@ -69,7 +69,7 @@ class VitGpt2ImageCaptioner(ImageCaptioner):
 		:raises Exception: If model initialization fails.
 		"""
 		import onnxruntime as ort
-		
+
 		# Load configuration file
 		try:
 			with open(configPath, "r", encoding="utf-8") as f:

--- a/source/_localCaptioner/captioner.py
+++ b/source/_localCaptioner/captioner.py
@@ -12,7 +12,6 @@ from abc import ABC, abstractmethod
 
 import numpy as np
 from PIL import Image
-import onnxruntime as ort
 
 from logHandler import log
 
@@ -69,6 +68,8 @@ class VitGpt2ImageCaptioner(ImageCaptioner):
 		:raises FileNotFoundError: If config file is not found.
 		:raises Exception: If model initialization fails.
 		"""
+		import onnxruntime as ort
+		
 		# Load configuration file
 		try:
 			with open(configPath, "r", encoding="utf-8") as f:

--- a/source/_localCaptioner/captioner.py
+++ b/source/_localCaptioner/captioner.py
@@ -68,6 +68,7 @@ class VitGpt2ImageCaptioner(ImageCaptioner):
 		:raises FileNotFoundError: If config file is not found.
 		:raises Exception: If model initialization fails.
 		"""
+		# Import late to avoid importing numpy at initialization
 		import onnxruntime as ort
 
 		# Load configuration file

--- a/source/_localCaptioner/imageDescriber.py
+++ b/source/_localCaptioner/imageDescriber.py
@@ -97,7 +97,7 @@ class ImageDescriber:
 		enable = config.conf["automatedImageDescriptions"]["enable"]
 		# Load model when initializing (may cause high memory usage)
 		if enable:
-			core.postNvdaStartup.register(self._loadModel)
+			core.postNvdaStartup.register(self.loadModelInBackground)
 
 	def terminate(self):
 		for t in [self.captionThread, self.loadModelThread]:
@@ -157,12 +157,12 @@ class ImageDescriber:
 		except Exception:
 			self.isModelLoaded = False
 			# Translators: error message when fail to load model
-			ui.message(pgettext("imageDesc", "failed to load image captioner"))
+			wx.CallAfter(ui.message, pgettext("imageDesc", "failed to load image captioner"))
 			log.exception("Failed to load image captioner model")
 		else:
 			self.isModelLoaded = True
 			# Translators: Message when successfully load the model
-			ui.message(pgettext("imageDesc", "image captioning on"))
+			wx.CallAfter(ui.message, pgettext("imageDesc", "image captioning on"))
 
 	def loadModelInBackground(self, localModelDirPath: str | None = None) -> None:
 		"""load model in child thread

--- a/source/_localCaptioner/imageDescriber.py
+++ b/source/_localCaptioner/imageDescriber.py
@@ -75,10 +75,10 @@ def _messageCaption(captioner: ImageCaptioner, imageData: bytes) -> None:
 		description = captioner.generateCaption(image=imageData)
 	except Exception:
 		# Translators: error message when an image description cannot be generated
-		ui.message(pgettext("imageDesc", "Failed to generate description"))
+		wx.CallAfter(ui.message, pgettext("imageDesc", "Failed to generate description"))
 		log.exception("Failed to generate caption")
 	else:
-		ui.message(description)
+		wx.CallAfter(ui.message, description)
 
 
 class ImageDescriber:

--- a/tests/system/robot/automatedImageDescriptions.robot
+++ b/tests/system/robot/automatedImageDescriptions.robot
@@ -16,12 +16,11 @@ Test Teardown	default teardown
 *** Keywords ***
 default teardown
 	${screenshotName}=	create_preserved_test_output_filename	failedTest.png
-	Run Keyword If Test Failed	Take Screenshot	${screenShotName}
+	Run Keyword If Test Failed	Take Screenshot	${screenshotName}
 	quit NVDA
 
 *** Test Cases ***
 automatedImageDescriptions
 	[Documentation]	Ensure that local captioner work
-	[Setup]	start NVDA	standard-doLoadMockModel.ini
 	NVDA_Caption	# run test
 

--- a/tests/system/robot/startupShutdownNVDA.robot
+++ b/tests/system/robot/startupShutdownNVDA.robot
@@ -17,7 +17,7 @@ Test Teardown	default teardown
 *** Keywords ***
 default teardown
 	${screenshotName}=	create_preserved_test_output_filename	failedTest.png
-	Run Keyword If Test Failed	Take Screenshot	${screenShotName}
+	Run Keyword If Test Failed	Take Screenshot	${screenshotName}
 	quit NVDA
 
 *** Test Cases ***

--- a/user_docs/en/changes.md
+++ b/user_docs/en/changes.md
@@ -20,7 +20,7 @@ Windows 10 on ARM is also no longer supported.
   * Press `NVDA+Windows+,` to get an AI generated image description. (#18475, @tianzeshi-study)
   * This is generated locally on the device - no information is sent to the internet.
   * A new unassigned command is available for quickly opening the settings dialog for local image description. (#18475)
-  * Another new unassigned command is available for toggle image captioning. (#18475)
+  * Another new unassigned command is available to toggle image captioning. (#18475)
 
 * Added the possibility to report when multiple items can be selected in a list control.
 This can be enabled using the "Report when lists support multiple selection" setting in NVDA's object presentation settings. (#18365 @LeonarddeR)

